### PR TITLE
User, Game classes on server

### PR DESCRIPTION
**Pull request by mrakhman**

### CODE REVIEW

File: src/server/controllers/Users.ts [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
+    const { id, username, room } = user;
+    this.id = id;
+    this.username = username;
+    this.room = room;
+    // if room exists ( === already has leader) - false
+    // if room doesn't exist - true
+    this.isLeader = false; // TODO: apply logic
+    users.push(this);
+
+    // if not username or room - return error ?
```
Comment (rizky) [inline src/server/controllers/Users.ts:20]: it's mandatory in type so no need

File: src/server/controllers/Users.ts [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+  isLeader: boolean;
+  
+  constructor(user: UserType) {
+    const { id, username, room } = user;
+    this.id = id;
+    this.username = username;
+    this.room = room;
+    // if room exists ( === already has leader) - false
+    // if room doesn't exist - true
+    this.isLeader = false; // TODO: apply logic
```
Comment (rizky) [inline src/server/controllers/Users.ts:17]: leader is the first user?

File: src/server/controllers/Users.ts (hunk 1 above)
Comment (rizky) [reply src/server/controllers/Users.ts:20]: need checking in Game class. to only create User if the username is not taken. if username exists send back error

File: src/server/controllers/Users.ts [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+    users.push(this);
+
+    // if not username or room - return error ?
+  }
+
+}
 
 // Get current user
-export const getCurrentUser = (id: string): Maybe<User> => {
+export const getCurrentUser = (id: string): Maybe<UserType> => {
```
Comment (rizky) [inline src/server/controllers/Users.ts:26]: put this in Game class?

File: src/server/controllers/Users.ts (hunk 2 above)
Comment (mrakhman) [reply src/server/controllers/Users.ts:17]: yes

File: src/server/models/Game.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    this.room = room;
+    games.push(this);
+  }
+
+  getUsers(): UserType[] {
+    return this.users;
+  }
+
+  // If game exists then room name is taken and user can only join the game
+  // Should it be not a method of this class?
```
Comment (rizky) [inline src/server/models/Game.ts:19]: Should we create 3 classes

Game 1 ---- * Room 1 ------- * User

Comment (mrakhman) [reply src/server/models/Game.ts:19]: 3 or more
what "----- *" means?

Comment (rizky) [reply src/server/models/Game.ts:19]: 1 ----- * is relationship one to many 

one game has many rooms

Comment (mrakhman) [reply src/server/models/Game.ts:19]: I'm not sure about "1 game many rooms", I thought 1 game = 1 room

Can you explain?

Comment (rizky) [reply src/server/models/Game.ts:19]: Game here is like the main controller. they can be many rooms playing in the same time

Comment (mrakhman) [reply src/server/models/Game.ts:19]: ok, I thought about having Games = Game[], but we can do Game = Room[]

**Comment by mrakhman** [comment]
Duplicate with conflicts, merged same code from userClass-2

**Pull request closed without merging**